src/algorithm/translate.rs

```diff
@@ -0,0 +1,196 @@
+use num_traits::{Float, FromPrimitive};
+use types::{Point, Polygon, LineString, Line, MultiPoint, MultiPolygon, MultiLineString};
+
+// translate a slice of points by the given offsets
+fn translation_matrix<T>(x_offset: T, y_offset: T, points: &[Point<T>]) -> Vec<Point<T>>
+    where T: Float
+{
+    points
+        .iter()
+        .map(|point| {
+                 let new_x = point.x() + x_offset;
+                 let new_y = point.y() + y_offset;
+                 Point::new(new_x, new_y)
```

Comment:
I'm not sure I follow, but if @rory does, I'll just wait on the PR!
IMO `rotate` / `translate` / `skew` / `scale` should all be implemented on top of a more general `affine` impl that populates an affine transformation matrix with the relevant values depending on the transformation type, but that would require us to pull in a linear algebra crate…